Service support implementation

shin- opened the pull request:
Services and tasks.

See: #1085

AkihiroSuda commented:
Seems need to apply some fixes:
e.g.
https://github.com/docker/engine-api/commit/14c70cc34bfee8f1335137b77cc8d496d0169bf8
https://github.com/docker/engine-api/commit/2d2ff7c7d6ee8bc7408e7c2440a4da1cf309c198

pacoxu commented:
Any update on this? 

we get error during update service
{"message":"Invalid service version '': strconv.ParseUint: parsing \"\": invalid syntax"}
as https://forums.docker.com/t/docker-swarm-scale-service-using-update-api/19589

atakada commented:
Is there anyone still working on it? I found some parameter names to be updated, but I'll be happy if there are any branches I can try.
https://github.com/atakada/docker-py/tree/service-support-fix

pacoxu commented:
@atakada 
Finally, we found that you have to use the update API like
POST /services/<service_id>/update?version=<current-version-index>
{
 "Name": "**"
"TaskTemplate" : {***}
}

Hope it can help you.

shin- commented:
Thank you for your help, folks! I've been focused on other tasks lately but I should be able to pick this back up soon.

atakada commented:
Thanks for comments&updates! Unfortunately I'm working for other stuff, but will come back.

shin- commented:
- I have rebased this on top of #1147 so I could use the swarm init methods in the integration tests.
- Updated methods and added the missing `update_service`
- Moved the `TaskTemplate`, `ContainerSpec` and similar classes to the `docker.types` package.
- Started writing docs. API docs still need to be filled out.

At that point everything should work properly, but please leave a comment if you're testing this and running into issues. :)

dnephin commented on tests/integration/service_test.py:
```diff
@@ -0,0 +1,190 @@
+import random
+
+import docker
+# import pytest
```
commented out import

dnephin commented on docker/types/services.py:
```diff
@@ -0,0 +1,181 @@
+
+    @classmethod
+    def parse_mount_string(cls, string):
+        parts = string.split(':')
```
The CLI doesn't support this syntax for mounts, so I'm not sure it should be supported here

dnephin commented:
LGTM

shin- commented on setup.py:
```diff
@@ -9,7 +9,7 @@
 
 requirements = [
-    'requests >= 2.5.2',
+    'requests >= 2.5.2, < 2.11',
```
https://github.com/docker/docker-py/pull/1204

glensc replied:
hmm, #1204 just removes the pin. so it's actually ok to use requests-2.11 without any other code changes?

shin- replied:
You can use any version >= 2.5.2, except 2.11.0. 
2.11.1 fixed the bug so that version or above is fine as well.

glensc replied:
thanks for clarification!
